types/mdast-util-toc-tests.ts

```diff
@@ -0,0 +1,80 @@
+import {Parent} from 'unist'
+
+import unified = require('unified')
+import u = require('unist-builder')
+import is = require('unist-util-is')
+import toc = require('mdast-util-toc')
+
+interface Link extends Parent {
+  type: 'link'
+}
+
+interface Paragraph extends Parent {
+  type: 'paragraph'
+}
+
+interface List extends Parent {
+  type: 'list'
+}
+
+interface ListItem extends Parent {
+  type: 'listItem'
+}
+
+const tree = u('root', [
+  u('heading', {depth: 1}, [u('text', 'Alpha')]),
+  u('heading', {depth: 2}, [u('text', 'Bravo')]),
+  u('heading', {depth: 3}, [u('text', 'Charlie')]),
+  u('heading', {depth: 2}, [u('text', 'Delta')])
+])
+
+const {map} = toc(tree)
+
+if (is<List>(map, 'list')) {
+  const [firstListItem] = map.children
+
+  if (is<ListItem>(firstListItem, 'listItem')) {
+    const [firstParagraph] = firstListItem.children
+
+    if (is<Paragraph>(firstParagraph, 'paragraph')) {
+      const [firstLink] = firstParagraph.children
+
+      is<Link>(firstLink, 'link')
```

Comment:
This is how the table of contents is generated by the util, but I'm not sure if this covers all cases or not?

Reply:
As far as I know this covers the main use case. (maybe @wooorm could weigh in if open of the options could change this structure)
From a cursory read through I don't see any major risks ever if more sub-types could be provided.
Folks could still use `is<NodeType>` to narrow the type as needed.

Reply:
I believe that options do not change the structure of the tree, except for the `tight` field on nodes.